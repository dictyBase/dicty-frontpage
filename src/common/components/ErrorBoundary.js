import React, { Component } from "react"
import Grid from "@material-ui/core/Grid"
import { withStyles } from "@material-ui/core/styles"

const styles = (theme) => ({
  gridContainer: {
    marginTop: "33px",
  },
  innerContainer: {
    textAlign: "left",
    paddingLeft: "10px",
    paddingRight: "10px",
    paddingBottom: "10px",
    backgroundColor: "#eff8fb",
    borderRadius: "15px",
    marginBottom: "10px",
    maxHeight: "440px",
    overflow: "auto",
    "@media (max-width: 768px)": {
      height: "350px",
    },
  },
})

/**
 * This is an ErrorBoundary wrapper that catches any
 * JavaScript errors and provides a fallback UI.
 */

class ErrorBoundary extends Component {
  constructor(props) {
    super(props)
    this.state = { error: null, errorInfo: null }
  }

  componentDidCatch(error, errorInfo) {
    // catch errors in any components below and re-render with error message
    this.setState({
      error,
      errorInfo,
    })
  }

  render() {
    const { errorInfo, error } = this.state
    const { children, classes } = this.props

    if (errorInfo) {
      // error path
      return (
        <Grid className={classes.gridContainer} container justify="center">
          <Grid item xs={6}>
            <div className={classes.innerContainer}>
              <center>
                <h2>Sorry! There was an error loading this page.</h2>
                <p>Something went wrong behind the scenes.</p>
                <em>{error && error.toString()}</em>
                <p>
                  If the problem persists, please email us at{" "}
                  <a href="mailto:dictybase@northwestern.edu">
                    dictybase@northwestern.edu
                  </a>
                  .
                </p>
              </center>
            </div>
          </Grid>
        </Grid>
      )
    }
    // normally, just render children
    return children
  }
}

export default withStyles(styles)(ErrorBoundary)
