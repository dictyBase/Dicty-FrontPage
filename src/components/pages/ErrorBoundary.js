import React from "react"
import Grid from "@material-ui/core/Grid"
import { withStyles } from "@material-ui/core/styles"
import { PaperContainer } from "styles"

// eslint-disable-next-line
const styles = theme => ({
  gridContainer: {
    marginTop: "50px",
  },
})

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props)
    this.state = { error: null, errorInfo: null }
  }

  componentDidCatch(error, errorInfo) {
    // catch errors in any components below and re-render with error message
    this.setState({
      error,
      errorInfo,
    })
  }

  render() {
    const { errorInfo, error } = this.state
    const { children, classes } = this.props

    if (errorInfo) {
      // error path
      return (
        <Grid className={classes.gridContainer} container justify="center">
          <Grid item xs={6}>
            <PaperContainer>
              <center>
                <h2>Sorry! There was an error loading this page.</h2>
                <p>Something went wrong with our user interface.</p>
                <em>{error && error.toString()}</em>
                <p>
                  If the problem persists, please email us at{" "}
                  <a href="mailto:dictybase@northwestern.edu">
                    dictybase@northwestern.edu
                  </a>
                  .
                </p>
              </center>
            </PaperContainer>
          </Grid>
        </Grid>
      )
    }
    // normally, just render children
    return children
  }
}

export default withStyles(styles)(ErrorBoundary)
