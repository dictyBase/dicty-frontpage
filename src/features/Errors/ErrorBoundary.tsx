import React, { Component } from "react"
import Grid from "@material-ui/core/Grid"
import { withStyles, Theme } from "@material-ui/core/styles"
import sadDicty from "common/assets/sad-dicty.png"

const styles = (theme: Theme) => ({
  gridContainer: {
    marginTop: "33px",
  },
  paper: {
    paddingTop: "10px",
    paddingBottom: "10px",
    backgroundColor: "#eff8fb",
    borderRadius: "15px",
    marginBottom: "10px",
    maxHeight: "500px",
    overflow: "auto",
    [theme.breakpoints.down("md")]: {
      height: "350px",
    },
  },
})

type Props = {
  /** Material-UI styling */
  classes: {
    gridContainer: string
    paper: string
  }
  /** Any children to render */
  children: any
}

type State = {
  /** If there is an error with JS code */
  hasError: boolean
}

/**
 * This is an ErrorBoundary wrapper that catches any
 * JavaScript errors and provides a fallback UI.
 * https://reactjs.org/docs/error-boundaries.html
 */

class ErrorBoundary extends Component<Props, State> {
  state = { hasError: false }

  static getDerivedStateFromError(error: Error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true }
  }

  componentDidCatch(error: Error, errorInfo: object) {
    console.error(error, errorInfo)
  }

  render() {
    const { hasError } = this.state
    const { children, classes } = this.props

    if (hasError) {
      return (
        <Grid className={classes.gridContainer} container justify="center">
          <Grid item xs={6} className={classes.paper}>
            <div style={{ textAlign: "center" }}>
              <img src={sadDicty} alt="Sad Dicty Logo" />
              <h2>Sorry! There was an error loading this page.</h2>
              <p>Something went wrong behind the scenes.</p>
              <p>
                If the problem persists, please email us at{" "}
                <a href="mailto:dictybase@northwestern.edu">
                  dictybase@northwestern.edu
                </a>
                .
              </p>
            </div>
          </Grid>
        </Grid>
      )
    }
    // normally, just render children
    return children
  }
}

export default withStyles(styles)(ErrorBoundary)
