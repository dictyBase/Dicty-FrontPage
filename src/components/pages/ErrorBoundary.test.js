import React from "react"
import toJson from "enzyme-to-json"
import { shallow } from "enzyme"
import "jest-styled-components"
import ErrorBoundary from "./ErrorBoundary"

test("matching a snapshot of ErrorBoundary", () => {
  const props = {
    children: {},
    classes: {},
  }
  const subject = shallow(<ErrorBoundary {...props} />)
  expect(toJson(subject)).toMatchSnapshot()
})
