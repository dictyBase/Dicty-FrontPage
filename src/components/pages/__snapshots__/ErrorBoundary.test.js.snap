// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of ErrorBoundary 1`] = `
<ErrorBoundary
  classes={
    Object {
      "gridContainer": "ErrorBoundary-gridContainer-1",
    }
  }
>
  <Component />
</ErrorBoundary>
`;
